### DISCUSSION TITLE
No connection to a WPA3-SAE / WPA2-PSK WiFi network

### DISCUSSION THREAD

**Discussion by wim04** [rern/rAudio#1851] [Fixed Issues]
Does nog connect tot a Odido Zyxel T56 modem.

**Comment by markblinkhorn** [comment]
Do you really need WPA3 ? - If not just reconfigure the AP to use WPA2-PSK only. The limitation seems to be within aarch64 and not necessarily an issue with rAudio.

**Reply by wim04** [reply]
Unfortunately I can only choose between WPA3-SAE / WPA2-PSK and WPA3.

**Comment by markblinkhorn** [comment]
I found a forum article here [https://bbs.archlinux.org/viewtopic.php?id=292059](url) which may give you a way forward. I haven't confimed it myself as I haven't enabled WPA3 on my setup. Good Luck.

**Comment by wim04** [comment]
Thx. Mark.
I found:
Mixed WPA2-PSK/WPA3-SAE access points will require an alternative setting for key_mgmt as shown below:

ssid="network SSID"
key_mgmt=WPA-PSK-SHA256
psk=xxx
ieee80211w=2

But i have to little knowledge to go on.

**Comment by markblinkhorn** [comment]
You need to modify /usr/lib/netctl/wpa - look for the section that looks like this:

```shell
      wpa)
        echo "key_mgmt=WPA-PSK SAE"
        echo "ieee80211w=1"
        if [[ "${#Key}" -eq 64 && $Key == +([[:xdigit:]]) ]]; then
            echo "psk=$Key"
        else
            echo "psk=$(wpa_quote "$Key")"
        fi
  
``` 

(was around line 230 onwards in mine) and edit the two `echo` lines to match your requirements - e.g. 

```shell
      wpa)
        echo "key_mgmt=WPA-PSK-SHA256 SAE"
        echo "ieee80211w=2"
        if [[ "${#Key}" -eq 64 && $Key == +([[:xdigit:]]) ]]; then
            echo "psk=$Key"
        else
            echo "psk=$(wpa_quote "$Key")"
        fi
  
``` 

The `ssid` and `psk` values come from your netctl profile (which you should already have set up from previously) but you can check by using:

`netctl list`
to give you a list of profiles stored on the system, and then:

`netctl edit <profile>`
if you need to modify it.

As I have already mentioned, I haven't done this on any of my systems as I am happy to use WPA2 for the time being. Backup as you go and you should be able to recover if it all turns out messy.

Good luck!
Mark

**Comment by wim04** [comment]
Hi Mark, Unfortunately no success. I have enabled the guest Wifi and there I can also choose WPA2 as security. This works. Problem solved. Thank you for thinking along.
